pkg/consuldp/lifecycle.go

```diff
@@ -144,9 +144,18 @@ func (m *lifecycleConfig) lifecycleServerExited() <-chan struct{} {
 	return m.errorExitCh
 }
 
-// gracefulShutdown blocks until shutdownGracePeriodSeconds seconds have elapsed, and, if
+func (m *lifecycleConfig) gracefulShutdownHandler(rw http.ResponseWriter, _ *http.Request) {
+	// Kick off graceful shutdown in a separate goroutine to avoid blocking
+	// sending an HTTP response
+	go m.gracefulShutdown()
+
+	// Return HTTP 200 Success
+	rw.WriteHeader(http.StatusOK)
+}
```

Comment:
@pglass Does this feel adequate to address https://github.com/hashicorp/consul-dataplane/pull/115#discussion_r1212254465?

Reply:
Yeah, I'm fine with this given your explanation there 🙂